From 46cabfa4567b23711282394f68d746649f290734 Mon Sep 17 00:00:00 2001
From: Abdullah Yousufi <abdullah.yousufi@cloudera.com>
Date: Wed, 15 Jun 2016 10:29:56 -0500
Subject: [PATCH 0623/1164] CDH-26348: HIVE-13987: Clarify current error shown
 when HS2 is down (Abdullah Yousufi, reviewed by
 Sergio Pena)

Change-Id: I43db0d8e37c62cfb1f88d3eec409bb22aed3b65f
---
 .../src/java/org/apache/hive/beeline/BeeLine.java  |    5 +++++
 beeline/src/main/resources/BeeLine.properties      |    2 ++
 2 files changed, 7 insertions(+)

diff --git a/beeline/src/java/org/apache/hive/beeline/BeeLine.java b/beeline/src/java/org/apache/hive/beeline/BeeLine.java
index d580ca8..8050263 100644
--- a/beeline/src/java/org/apache/hive/beeline/BeeLine.java
+++ b/beeline/src/java/org/apache/hive/beeline/BeeLine.java
@@ -90,6 +90,7 @@
 import org.apache.commons.cli.ParseException;
 import org.apache.hadoop.io.IOUtils;
 import org.apache.hive.beeline.cli.CliOptionsProcessor;
+import org.apache.thrift.transport.TTransportException;
 
 /**
  * A console SQL shell with command completion.
@@ -1684,6 +1685,10 @@ void handleSQLException(SQLException e) {
       return;
     }
 
+    if (e.getCause() instanceof TTransportException) {
+      error(loc("hs2-unavailable"));
+    }
+
     error(loc(e instanceof SQLWarning ? "Warning" : "Error",
         new Object[] {
             e.getMessage() == null ? "" : e.getMessage().trim(),
diff --git a/beeline/src/main/resources/BeeLine.properties b/beeline/src/main/resources/BeeLine.properties
index ad7280d..7bf6ec1 100644
--- a/beeline/src/main/resources/BeeLine.properties
+++ b/beeline/src/main/resources/BeeLine.properties
@@ -140,6 +140,8 @@ active-connections: 0#No active connections|1#{0} active connection:|1<{0} activ
 
 time-ms: ({0,number,#.###} seconds)
 
+hs2-unavailable: HS2 may be unavailable, check server status
+
 cmd-usage: Usage: java org.apache.hive.cli.beeline.BeeLine \n \
 \  -u <database url>               the JDBC URL to connect to\n \
 \  -n <username>                   the username to connect as\n \
-- 
1.7.9.5

